### PR TITLE
Judge-specific text display

### LINKED ISSUE #736: Be able to show arbitrary text (including hyperlinks) in every workspace of a tree
The current primary motivation for this is to be able to link to a doc that specifies what strategy judges should use on the tree.

But we may want it for other things in the future.

We could do this as an extension of https://github.com/oughtinc/mosaic/issues/626 -- I'd be totally fine with something hacky like:
1. All text in the root-level stratchpad gets shown to experts
2. All text between, say < and > also gets shown to judges

### PR DESCRIPTION
**Pull request by andrewschreiber**
closes #736

Wrap scratchpad text in `<judge> judge text here </judge>` for judge-specific text. Experts will be able to see the judge text, but judges wont be able to see anything outside the judge text.

### CODE REVIEW

**Comment by andrewschreiber** [comment]
Tested and looking as expected.

Expert View:
image

Judge View:
image

**Comment by andrewschreiber** [comment]
Realizing we should probably document this new functionality somewhere. Thoughts? @brachbach @Mati-Roy

**Comment by brachbach** [comment]
1. I'd do `<judge>...</judge>` instead of `<judge>...<judge>`, for consistency with HTML
2. I think would make sense to document in the `mosaic-api-scripts` README under the section about creating experiments/trees (since we mainly expect to use the functionality via the API)

File: client/src/pages/EpisodeShowPage/index.tsx [hunk 1]
```diff
@@ -677,6 +678,17 @@ export class WorkspaceView extends React.Component<any, any> {
       ? this.getAvailablePointers(workspace, rootWorkspaceScratchPad)
       : this.getAvailablePointers(workspace);
 
+    const shouldShowJudgeTextFromScratchpad =
+      hasParent &&
+      !isOracleWorkspace &&
+      doesRootWorkspaceScratchpadContainRelevantContent;
+
+    const judgeRegex = /(?<=<judge>)(.*)(?=<judge>)/;
```
Comment (zjmiller) [inline client/src/pages/EpisodeShowPage/index.tsx]: Lookbehind `?<=` is not supported by Firefox, so Firefox crashes when loading a workspace.

I'm pretty sure you can change this regex to `/<judge>(.*)<judge>/`. And then the second item in the result array returned by `.exec` will be the match within the parentheses (i.e., the text in between the tags). https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec

File: client/src/pages/EpisodeShowPage/index.tsx [hunk 2]
```diff
@@ -677,6 +678,17 @@ export class WorkspaceView extends React.Component<any, any> {
...
+    const judgeTextArray = judgeRegex.exec(
+      listOfSlateNodesToText(_.get(rootWorkspaceScratchPad, "value"), false),
+    );
+    const judgeText = judgeTextArray ? judgeTextArray[0] : null;
```
Comment (zjmiller) [inline client/src/pages/EpisodeShowPage/index.tsx]: So if you change the regex as mentioned above, the `judgeTextArray[0]` should be `judgeTextArray[1]`.

**Comment by Mati-Roy** [comment]
what use case did we develop this functionality for?

**Comment by andrewschreiber** [comment]
The use case is inline communication of specific strategies that we want the participants to use, according to @brachbach

File: client/src/pages/EpisodeShowPage/index.tsx (hunk 1 above)
Comment (andrewschreiber) [reply client/src/pages/EpisodeShowPage/index.tsx]: Good call, will update. That regex would work, according to https://regexr.com/

**Comment by andrewschreiber** [comment]
Updated regex to work on Firefox and terminate with </judge>
image

Will make a mosaic-api-scripts PR with the documentation after this is merged.

Ready for 2nd review pass @zjmiller

File: client/src/pages/EpisodeShowPage/index.tsx [hunk 3]
```diff
@@ -683,11 +683,11 @@ export class WorkspaceView extends React.Component<any, any> {
       !isOracleWorkspace &&
       doesRootWorkspaceScratchpadContainRelevantContent;
 
-    const judgeRegex = /(?<=<judge>)(.*)(?=<judge>)/;
+    const judgeRegex = /<judge>(.*)<\/judge>/;
```
Comment (zjmiller) [inline client/src/pages/EpisodeShowPage/index.tsx]: Wait, do we want `<judge><judge>` or `<judge></judge>`?

When I first reviewed it was the former, but now it's the latter.

Comment (andrewschreiber) [reply client/src/pages/EpisodeShowPage/index.tsx]: The later - per Ben's request / reference to HTML

**Pull request merged**